os/kernel/group/group_signal.c

```diff
@@ -240,7 +240,11 @@ int group_signal(FAR struct task_group_s *group, FAR siginfo_t *siginfo)
 	 * nothing if were were called from an interrupt handler).
 	 */
 
+#ifdef CONFIG_SMP
+	irqstate_t flags = enter_critical_section()
+#else
 	sched_lock();
+#endif
```

Comment:
Sched_lock and irqsave are different. Do you need to replace sched_lock to enter_critical_section in this case?

Reply:
As i understand, sched lock only disables scheduler. But irqsave disables all irq. 
In case smp is enabled, we need to disable all irq in some cases. Hence this is the code according to nuttx.